src/dependencies.js

```diff
@@ -84,33 +84,66 @@ const getFileDependencies = async function(path, packageJson, state) {
   const dependencies = precinct.paperwork(path, { includeCore: false })
 
   const depsPaths = await Promise.all(
-    dependencies.map(dependency => getImportDependencies(dependency, basedir, packageJson, state))
+    dependencies.map(dependency =>
+      getImportDependencies({ dependency, basedir, packageJson, state, treeShakingModule })
+    )
   )
   return [].concat(...depsPaths)
 }
 
-// `require()` statements can be either `require('moduleName')` or
-// `require(path)`
-const getImportDependencies = function(dependency, basedir, packageJson, state) {
-  if (LOCAL_IMPORT_REGEXP.test(dependency)) {
-    return getLocalImportDependencies(dependency, basedir, packageJson, state)
+const isTreeShakingModule = function({ dependency, treeShakingModule }) {
+  return treeShakingModule === getModuleName(dependency)
+}
+
+const getImportDependencies = function({ dependency, basedir, packageJson, state, treeShakingModule }) {
+  if (LOCAL_IMPORT_REGEXP.test(dependency) || isTreeShakingModule({ dependency, treeShakingModule })) {
+    return getTreeShakedDependencies({ dependency, basedir, packageJson, state, treeShakingModule })
+  }
+
+  if (isTreeShakable(dependency)) {
+    const module = getModuleName(dependency)
+    return getTreeShakedDependencies({ dependency, basedir, packageJson, state, treeShakingModule: module })
   }
```

Comment:
[sand] Would the following be simpler?

```js
const isTreeShakingModule = function(moduleName, treeShakingModule) {
  return treeShakingModule === moduleName
}

const getImportDependencies = function({ dependency, basedir, packageJson, state, treeShakingModule }) {
  const moduleName = getModuleName(dependency)
  if (LOCAL_IMPORT_REGEXP.test(dependency) || isTreeShakingModule(moduleName, treeShakingModule) || isTreeShakable(dependency)) {
    return getTreeShakedDependencies({ dependency, basedir, packageJson, state, treeShakingModule: moduleName })
  }

  return getAllDependencies(dependency, basedir, state, packageJson)
}
```

Reply:
Sure, I'm good either way

Reply:
Update let me just check passing `moduleName` for local imports doesn't break anything

Reply:
Update; It does break as it stops tree shaking when switching between local imports inside a `next/*` `require`

Reply:
Is this https://github.com/netlify/zip-it-and-ship-it/pull/242/commits/d4edf4930de0f8f2f48e52fb5cdf159006a82f1a better?

Reply:
Yes, this works! :+1: